### PR TITLE
Issue 14672 filter host actions

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
fix for https://github.com/dotCMS/core/issues/14628 is included here as well.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java [hunk 1]
```diff
@@ -117,32 +117,41 @@ public BulkActionView findBulkActions(final User user,
...
-        Logger.debug(getClass(),()-> "esSearch: "+ query);
-        return this.buildBulkActionView(results, user);
+            final User user) throws DotDataException {
+
+        try {
+            final String prepareBulkActionsQuery = LuceneQueryUtils
+                    .prepareBulkActionsQuery(luceneQuery);
+
+            //We should only be considering Working content.
+            final String query = "{\n"
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java]: shouldn't be a constant?

File: dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java [hunk 2]
```diff
@@ -117,32 +117,41 @@ public BulkActionView findBulkActions(final User user,
...
+            final String preparedQuery = String.format(query, prepareBulkActionsQuery);
+
+            final SearchResponse response = this.contentletAPI
+                    .esSearchRaw(preparedQuery.toLowerCase(), false, user, false);
+
+            Logger.debug(getClass(), () -> "luceneQuery: " + prepareBulkActionsQuery);
+            Logger.debug(getClass(), () -> "esSearch: " + query);
+            return this.buildBulkActionView(response, user);
+        } catch (Exception e) {
+            throw new DotDataException(e);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java]: Log me

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ContentletUtil.java [hunk 3]
```diff
@@ -44,6 +45,11 @@ public static boolean isFieldTypeAllowedOnImportExport(final Field field){
 		return !fieldTypesToExcludeFromImportExport.contains(field.getFieldType());
 	}
 
+	public static boolean isHost(final Contentlet contentlet){
+		final Structure hostStrucuture = CacheLocator.getContentTypeCache().getStructureByVelocityVarName("Host");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ContentletUtil.java]: Host.HOST_VELOCITY_VAR_NAME there is a constant for Host

File: dotCMS/src/main/java/com/dotmarketing/util/LuceneQueryUtils.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+public class LuceneQueryUtils {
+
+
+    /**
+     *
+     * @param luceneQuery
+     * @return
+     */
+    private static String removeQueryPrefix(final String luceneQuery){
+        final String cleanedUpQuery;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/LuceneQueryUtils.java]: you can use return (luceneQuery.startsWith("query_"))?luceneQuery.replace( "query_", StringPool.BLANK): luceneQuery;

it is just a style, up to you

File: dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java (hunk 2 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java]: The exception bubbles up and it gets logged by the entry point method on the WebResource. So its currently getting logged. 

Adding logs like that adds noise in the log files making the exception appear twice.

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ContentletUtil.java (hunk 3 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ContentletUtil.java]: done.

File: dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java]: done.

File: dotCMS/src/main/java/com/dotmarketing/util/LuceneQueryUtils.java (hunk 4 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/util/LuceneQueryUtils.java]: done.

**Review by jdotcms — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**